gomock/matchers.go

```diff
@@ -280,6 +292,15 @@ func All(ms ...Matcher) Matcher { return allMatcher{ms} }
 // Any returns a matcher that always matches.
 func Any() Matcher { return anyMatcher{} }
 
+// Cond returns a specialized matchers customizable for complex matching behaviour.
+// This is particularly useful in case you want to match over a field of a custom struct, or dynamic logic.
```

Comment:
I think we can be a little more precise with what this matcher does.
Something like:

```
// Cond returns a matcher that matches when the given function returns true
// after passing it the parameter to the mock function.
```

Reply:
Updated 💪🏻